Attempt to improve CPU use by declaring things const

mlyle opened the pull request:
Also just assume systime is milliseconds because the math before was faulty.

May improve #2200

mlyle committed:
uavobjectmanager: make callbacks use const msg

Also avoid a copy of msg in event pumping.

mlyle committed:
uavobjectmanager: additional consting

glowtape commented:
L1690 in uavobjectmanager.c needs const.

Gives slight improvements, still not as far as reverting those commits mentioned in #2200.

mlyle committed:
pios_thread: avoid 64 bit math in time paths

Note this requires that the frequency be 1000.  Optimizer couldn't
suppress the math because of edge case when time near 0.

mlyle commented:
OK, one more thing to try in here.

mlyle commented:
I'd advocate for this to merge-- with the PIOS_Thread time change we're back to where we were originally in CPU use.
